server/matchmaker/matchmaker_queue.py

```diff
@@ -37,18 +38,37 @@ def __exit__(self, exc_type, exc_value, traceback):
 class MatchmakerQueue:
     def __init__(
         self,
-        queue_name: str,
-        game_service: "GameService"
+        game_service: "GameService",
+        name: str,
+        name_key: str,
+        map_pools: List[Tuple[MapPool, Optional[int], Optional[int]]] = []
     ):
         self.game_service = game_service
-        self.queue_name = queue_name
+        self.name = name
+        self.name_key = name_key
+        self.map_pools = {info[0].id: info for info in map_pools}
 
         self.queue: Dict[Search, Search] = OrderedDict()
         self._matches: Deque[Match] = deque()
         self._is_running = True
 
-        self.timer = PopTimer(self.queue_name)
-        self._logger.debug("MatchmakerQueue initialized for %s", queue_name)
+        self.timer = PopTimer(self.name)
+
+    def add_map_pool(
+        self,
+        map_pool: MapPool,
+        min_rating: Optional[int],
+        max_rating: Optional[int]
+    ):
+        self.map_pools[map_pool.id] = (map_pool, min_rating, max_rating)
+
+    def get_map_pool_for_rating(self, rating: int) -> Optional[MapPool]:
+        for map_pool, min_rating, max_rating in self.map_pools.values():
+            if min_rating is not None and rating < min_rating:
+                continue
+            if max_rating is not None and rating > max_rating:
+                continue
+            return map_pool
```

Comment:
I guess there are some edge cases where what map pool you get depends on the order of `self.map_pools.values()`, but I think that is ok in the interest of being robust against people specifying rating bounds however they see fit.
e.g. if `(min_rating, max_rating)` is `[(1,2), (2,3)]` and your rating is exactly `2`.
Or of course if they overlap like `[(1,3),(2,3)]`.